tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+    ]
+
+    for sample, expected_variance in zip(samples, expected_variances):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.variance()),
+                          expected_variance)
+      self.assertAllClose(self.evaluate(dist.stddev()),
+                          np.sqrt(expected_variance))
+
+
+class EmpiricalVectorTest(test_util.VectorDistributionTestHelpers):
+
+  def testSamples(self):
+    for samples_shape in ([2, 4], [4, 2, 4], [2, 2, 2, 4]):
+      input_ = random_samples(samples_shape)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=samples_shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllClose(input_ph, self.evaluate(dist.samples))
+
+    invalid_samples = [
+        0,
+        [0, 1],
+    ]
+    for samples in invalid_samples:
+      with self.assertRaises(ValueError):
+        dist = empirical.Empirical(samples=samples, event_ndims=1)
+
+  def testShapes(self):
+    for samples_shape in ([2, 4], [4, 2, 4]):
+      input_ = random_samples(samples_shape)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=samples_shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllEqual(samples_shape[:-2],
+                          self.evaluate(dist.batch_shape_tensor()))
+      self.assertAllEqual(samples_shape[-1:],
+                          self.evaluate(dist.event_shape_tensor()))
+
+  def testCdfNoBatch(self):
+    sample = [[0, 0], [0, 1], [0, 1], [1, 2]]
+    events = [
+        [0, 1],
+        [[0, 1], [1, 2]],
+        [[[0, 0], [0, 1]],
+         [[1, 1], [1, 2]]]
+    ]
+    expected_cdfs = [
+        [0.75, 0.75],
+        [[0.75, 0.75],
+         [1, 1]],
+        [[[0.75, 0.25], [0.75, 0.75]],
+         [[1., 0.75], [1., 1.]]]
+    ]
+
+    for event, expected_cdf in zip(events, expected_cdfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllClose(self.evaluate(dist.cdf(event)),
+                          expected_cdf)
+      self.assertAllClose(self.evaluate(dist.log_cdf(event)),
+                          np.log(expected_cdf))
+
+  def testCdfWithBatch(self):
+    sample = [[[0, 0], [0, 1], [0, 1], [1, 1]],
+              [[0, 1], [2, 2], [2, 2], [2, 2]]]
+    events = [
+        [0, 1],
+        [[0, 0], [2, 2]],
+        [[[0, 0], [0, 1]],
+         [[0, 1], [2, 2]]]
+    ]
+    expected_cdfs = [
+        [[0.75, 1.],
```

Comment:
similar comment applies here. sample has batch shape [2], nsamples=4, event shape [2], so [2] is now the minimal shape of my expected output.
events[0] has scalar batch shape, so I expect shape [2] cdf out `[.75, .25]`. but instead I see a [2,2] coming out
events[1] has batch shape [2], which should align with the batch shape of samples, so I still expect a [2] shape cdf out `[.25, 1.0]`
events[2] has batch shape [2,2], so the inner [2] aligns with batch shape of samples, so I expect a [2,2] cdf out `[[.25, .25],[.75,1.0]]`

It would also be useful to test an event with shape [2,1] and verify the 1 gets broadcast against the batch shape [2] of the distribution for a [2,2] cdf out.